next.config.js

```diff
@@ -47,6 +62,7 @@ module.exports = withSwingset({ componentsRoot: 'src/components/*' })(
       ENABLE_VERSIONED_DOCS: process.env.ENABLE_VERSIONED_DOCS || false,
       IS_CONTENT_PREVIEW: process.env.IS_CONTENT_PREVIEW,
       DEV_IO: process.env.DEV_IO,
+      ...getHashiEnvironmentConfig(),
```

Comment:
I have a bit of trepidation about exposing a configuration as a JSON file that _could_ contain deeply-nested values when it's going to get spread into the `env`. What do you think about using `dotenv.parse()` to parse a ``${env}.env`` file instead (or ``.env.${env}`` if you wanted to be consistent with the naming convention used by Next)? That way environment variable configuration is consistent between data in `config/` and files like `.env.local`.

Reply:
We could add support for nested keys right off the bat by using something like [flat](https://replit.com/@dstaley/BuoyantExternalLicenses#index.js). If we come up with an alternative in the future that allows for traversing the object in user code, we could implement these side-by-side as a transition period.

Reply:
👍 on both fronts - I dig the possibility of having nested values in config, and dig `flat` as a solution to enable that. I like the idea of the nice safe feeling we'd have if we account for nesting off the bat - either by using `flat`, or by doing some validation after `JSON.parse` in `getHashiEnvironmentConfig` to ensure the config object isn't deeply nested.